### PR TITLE
Update workflows to point to `go.mod`

### PR DESCRIPTION
**Pull request by thepetk**
## What does this PR do?

Replaces the hardcoded go version in the workflows by fetching the go version directly from the `go.mod` file.

### Which issue(s) does this PR fix

N/A

### PR acceptance criteria

Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened.

- [ ] Unit/Functional tests

  

- [ ] Documentation

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -62,8 +62,8 @@ jobs:
       - name: Set up Go 1.x
         uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
         with:
-          go-version: 1.21
+          go-version: 'go.mod'
```
Comment (michael-valdron) [inline .github/workflows/CI.yml:65]:
```suggestion
          go-version-file: 'go.mod'
```

File: .github/workflows/check_registry.yaml [hunk 2]
```diff
@@ -27,7 +27,7 @@ jobs:
         id: setup-go
         uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
         with:
-          go-version: "1.21"
+          go-version: 'go.mod'
```
Comment (michael-valdron) [inline .github/workflows/check_registry.yaml:30]:
```suggestion
          go-version-file: 'go.mod'
```

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -33,7 +33,7 @@ jobs:
           fetch-depth: 0
       - uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
         with:
-          go-version: '1.21'
+          go-version: 'go.mod'
```
Comment (michael-valdron) [inline .github/workflows/release.yaml:36]:
```suggestion
          go-version-file: 'go.mod'
```

File: .github/workflows/CI.yml [hunk 4]
```diff
@@ -14,7 +14,7 @@ jobs:
       - name: Setup Go environment
         uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
         with:
-          go-version: '1.21'
+          go-version: 'go.mod'
```
Comment (michael-valdron) [inline .github/workflows/CI.yml:17]:
```suggestion
          go-version-file: 'go.mod'
```

**Review by michael-valdron — CHANGES_REQUESTED** [review]
These are causing the current check failures.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (thepetk) [reply .github/workflows/CI.yml:65]: Ouch! Something was messed up when I initially rebased locally.

**Review by Jdubrick — APPROVED** [review]
/lgtm 

As long as @michael-valdron likes the changes to his suggestions

**Review by michael-valdron — APPROVED** [review]
/lgtm

**Pull request merged**